components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/issuers/APIMTokenIssuer.java

```diff
@@ -50,22 +55,36 @@ public String accessToken(OAuthTokenReqMessageContext tokReqMsgCtx) throws OAuth
                 long output = end_time - start_time;
                 log.debug("Time taken to load the Application from database in milliseconds : " + output / 1000000);
             }
-
-            if (APIConstants.JWT.equals(application.getTokenType())) {
-                JwtTokenInfoDTO jwtTokenInfoDTO = APIMTokenIssuerUtil.getJwtTokenInfoDTO(application);
-                APIMJWTGenerator apimjwtGenerator = new APIMJWTGenerator();
-                if (log.isDebugEnabled()) {
-                    long end_time_2 = System.nanoTime();
-                    long output = end_time_2 - start_time;
-                    log.debug("Time taken to generate the JWG in milliseconds : " + output / 1000000);
+            if (application != null) {
+                String tokenType = application.getTokenType();
+                if (APIConstants.JWT.equals(tokenType)) {
+                    String[] scopes = tokReqMsgCtx.getScope();
+                    JwtTokenInfoDTO jwtTokenInfoDTO = APIMTokenIssuerUtil.getJwtTokenInfoDTO(application);
+                    jwtTokenInfoDTO.setScopes(scopes);
+                    jwtTokenInfoDTO.setExpirationTime(tokReqMsgCtx.getValidityPeriod());
+                    APIMJWTGenerator apimjwtGenerator = new APIMJWTGenerator();
+                    if (log.isDebugEnabled()) {
+                        long end_time_2 = System.nanoTime();
+                        long output = end_time_2 - start_time;
+                        log.debug("Time taken to generate the JWG in milliseconds : " + output / 1000000);
+                    }
+                    return apimjwtGenerator.generateJWT(jwtTokenInfoDTO);
                 }
-                return apimjwtGenerator.generateJWT(jwtTokenInfoDTO);
             }
 
         } catch (APIManagementException e) {
             log.error("Error occurred while getting JWT Token client ID : " + clientId, e);
             throw new OAuthSystemException("Error occurred while getting JWT Token client ID : " + clientId, e);
         }
-        return super.accessToken(tokReqMsgCtx);
+        return OAuthServerConfiguration.getInstance().getOAuthTokenGenerator().accessToken();
+    }
+
+    @Override
+    public String getAccessTokenHash(String accessToken) throws OAuthSystemException {
```

Comment:
can't the access token be null?

Reply:
shouldn't we throw an exception of the token is null?

Reply:
Here in our case token cannot be null.